server/data/data-library/src/main/java/org/apache/james/rrt/lib/CanSendFromImpl.java

```diff
@@ -50,6 +57,16 @@ public boolean userCanSendFrom(Username connectedUser, Username fromUser) {
         }
     }
 
+    @Override
+    public Stream<MailAddress> allValidFromAddressesForUser(Username user) throws RecipientRewriteTable.ErrorMappingException, RecipientRewriteTableException {
+        List<String> domains = relatedDomains(user).collect(Guavate.toImmutableList());
+
+        return relatedAliases(user)
+            .flatMap(userPart -> domains.stream()
+                .map(domainPart -> userPart + "@" + domainPart)
```

Comment:
Later refacoting: string concatenation spotted.

Shouldn't we have a domain method for this?